sdk/lib/html/dart2js/html_dart2js.dart

```diff
@@ -18134,14 +18134,17 @@ class HttpRequest extends HttpRequestEventTarget {
    *
    * See also: [authorization headers](http://en.wikipedia.org/wiki/Basic_access_authentication).
    */
-  static Future<HttpRequest> request(String url,
-      {String? method,
-      bool? withCredentials,
-      String? responseType,
-      String? mimeType,
-      Map<String, String>? requestHeaders,
-      sendData,
-      void onProgress(ProgressEvent e)?}) {
+  static Future<HttpRequest> request(
```

Comment:
Please modify `tools/dom/templates/html/impl/impl_XMLHttpRequest.darttemplate` instead of the file directly. You'll need to use `void onUploadProgress(ProgressEvent e)$NULLABLE` there instead of with `?`. These files were used to generate both legacy and null-safe versions of this library, hence the `$NULLABLE`.

`dart:html` is generated by using these templates, source files, and codegen scripts, so any edits in the file directly without modifying one of those will remove this code on regeneration. Once you've made your edits there you can run `go.sh` in `tools/dom/scripts` to generate the library.

Reply:
I run `./go.sh` then facing this error.

```
+ python2.7 ./dartdomgenerator.py --systems=htmldart2js --logging=40 --update-dom-metadata --gen-interop ''
Traceback (most recent call last):
  File "./dartdomgenerator.py", line 32, in <module>
    assert (os.path.exists(third_party_dir))
AssertionError
```

Reply:
Make sure you get SDK source code by following https://github.com/dart-lang/sdk/wiki/Building#getting-the-source - just cloning GitHub repository is not enough to make a proper development checkout.